src/toil/common.py

```diff
@@ -575,6 +582,38 @@ def addGroup(headingString, bodyString):
         raise RuntimeError("Unanticipated class passed to addOptions(), %s. Expecting "
                            "argparse.ArgumentParser" % parser.__class__)
 
+def checkValidNodeTypes(provisioner, nodeTypes):
```

Comment:
Azure and GCE already do this. This method should be in the AWSProvisioner.

Reply:
GCE doesn't seem to: #2135